ipa-core/src/net/server/mod.rs

```diff
@@ -244,37 +241,25 @@ where
 
 async fn certificate_and_key(
     config: &ServerConfig,
-) -> Result<(Vec<Certificate>, PrivateKey), BoxError> {
-    let (cert, key) = match &config.tls {
-        None => return Err("missing TLS configuration".into()),
+) -> Result<(Vec<CertificateDer<'static>>, PrivateKeyDer<'static>), BoxError> {
+    match &config.tls {
+        None => Err("missing TLS configuration".into()),
         Some(TlsConfig::Inline {
             certificate,
             private_key,
-        }) => (
-            Cow::Borrowed(certificate.as_bytes()),
-            Cow::Borrowed(private_key.as_bytes()),
-        ),
+        }) => {
+            parse_certificate_and_private_key_bytes(certificate.as_bytes(), private_key.as_bytes())
```

Comment:
I don't think you need to change anything here. This should work too and it is just a single line change

```rust
   let (cert, key) = ... // exactly as it was before
  
   parse_certificate_and_private_key_bytes(cert.as_ref(), key.as_ref()).map_err(BoxError::from)

```

Reply:
I also wanted to simplify the code. With my change, Cow doesn't add anything since `parse_certificate_and_private_key_bytes` just borrows

Reply:
I find the version that uses `Cow` easier to understand